Code/GraphMol/Chirality.cpp

```diff
@@ -769,6 +444,213 @@ const Atom *findHighestCIPNeighbor(const Atom *atom, const Atom *skipAtom) {
+  if (is_regular_h(*bondAtom)) {
+    hSeen = true;
+  }
+
+  unsigned int refIdx = mol.getNumBonds() + 1;
+  std::vector<RDGeom::Point3D> bondVects;
+  bool allSingle = true;
+  unsigned int nbrIdx = 0;
+  for (const auto nbrBond : mol.atomBonds(atom)) {
+    const auto oAtom = nbrBond->getOtherAtom(atom);
+    auto tmpPt = conf->getAtomPos(oAtom->getIdx());
+    if (nbrBond == bond) {
+      refIdx = nbrIdx;
+      tmpPt = refPt;
+    } else {
+      // theoretically we could confirm that this is a single bond,
+      // but it's not impossible that at some point in the future we
+      // could allow wedged multiple bonds for things like atropisomers
+      if (nbrBond->getBeginAtomIdx() == atom->getIdx() &&
+          nbrBond->getBondDir() == Bond::BondDir::BEGINWEDGE) {
+        tmpPt.z = pseudo3DOffset;
+      } else if (nbrBond->getBeginAtomIdx() == atom->getIdx() &&
+                 nbrBond->getBondDir() == Bond::BondDir::BEGINDASH) {
+        tmpPt.z = -pseudo3DOffset;
+      } else {
+        tmpPt.z = 0;
+      }
+    }
+    ++nbrIdx;
+    if (nbrBond->getBondType() != Bond::SINGLE) {
+      allSingle = false;
+    }
+    bondVects.push_back(centerLoc.directionVector(tmpPt));
+    if (is_regular_h(*oAtom)) {
+      hSeen = true;
+    }
+    neighborBondIndices.push_back(nbrBond->getIdx());
+  }
+  CHECK_INVARIANT(refIdx < mol.getNumBonds(),
+                  "could not find reference bond in neighbors");
+
+  auto nNbrs = bondVects.size();
+
+  //----------------------------------------------------------
+  //
+  //  Return now if there aren't at least 3 non-H bonds to the atom.
+  //  (we can implicitly add a single H to 3 coordinate atoms, but
+  //  we're horked otherwise).
+  //
+  //----------------------------------------------------------
+  if (nNbrs < 3 || nNbrs > 4 || (hSeen && nNbrs < 4)) {
+    return std::nullopt;
+  }
+
+  //----------------------------------------------------------
+  //
+  //  Continue if there are all single bonds or if we're considering
+  //  4-coordinate P or S
+  //
+  //----------------------------------------------------------
+  if (allSingle || atom->getAtomicNum() == 15 || atom->getAtomicNum() == 16) {
+    double vol;
+    unsigned int order[4] = {0, 1, 2, 3};
+    double prefactor = 1;
+    if (refIdx != 0) {
+      // bring the wedged bond to the front so that we always consider it
+      std::swap(order[0], order[refIdx]);
+      prefactor *= -1;
+    }
+
+    // three-coordinate special cases where chirality cannot be determined
+    //
+    //  Case 1:
+    //  this one is never allowed with different directions for the bonds to 1
+    //  and 2
+    //     0   2
+    //      \ /
+    //       C
+    //       *
+    //       1
+    //   This is ST-1.2.10 in the IUPAC guidelines
+    //
+    //  Case 2: all bonds are wedged in the same direction
+
+    if (nNbrs == 3) {
+      bool conflict = false;
+      if (bondVects[order[1]].z * bondVects[order[0]].z < -1e-4 &&
```

Comment:
1e-4 is used a lot, should this be a constant or at least split up between distance, angle and volume for clarity?